src/backend/cdb/cdbpartition.c

```diff
@@ -6784,6 +6784,22 @@ atpxPartAddList(Relation rel,
 	if (pelem->storeAttr)
 		ct->options = (List *) ((AlterPartitionCmd *) pelem->storeAttr)->arg1;
 
+	if (gp_add_partition_inherits_table_setting)
+	{
+		ListCell *lc;
+		List	 *res = ct->options;
+
+		foreach(lc, reloptions_list(RelationGetRelid(rel)))
+		{
+			DefElem *de = lfirst(lc);
+
+			if (!reloptions_has_opt(ct->options, de->defname))
+				res = lappend(res, de);
+		}
+
+		ct->options = res;
```

Comment:
`res` seems to be redundant
```suggestion

		foreach(lc, reloptions_list(RelationGetRelid(rel)))
		{
			DefElem *de = lfirst(lc);

			if (!reloptions_has_opt(ct->options, de->defname))
				ct->options = lappend(ct->options, de);
		}
```

Reply:
Have you tested this in practice?

Reply:
Yes, I have. All the tests have passed